dev/core/src/com/google/gwt/dev/shell/StaticResourceServer.java

```diff
@@ -0,0 +1,190 @@
+import com.google.gwt.core.ext.TreeLogger;
+import com.google.gwt.core.ext.UnableToCompleteException;
+import com.google.gwt.dev.shell.jetty.ClientAuthType;
+import com.google.gwt.dev.shell.jetty.JettyLauncherUtils;
+import com.google.gwt.dev.shell.jetty.JettyTreeLogger;
+import com.google.gwt.dev.shell.jetty.SslConfiguration;
+import org.eclipse.jetty.security.ConstraintSecurityHandler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ErrorPageErrorHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+import java.io.File;
+import java.util.Optional;
+
+/**
+ * Simple webserver that only hosts static resources. Intended to be the replacement
+ * for JettyLauncher.
+ */
+public class StaticResourceServer extends ServletContainerLauncher {
+
+  private String bindAddress = null;
+
+  private SslConfiguration sslConfig = new SslConfiguration(ClientAuthType.NONE, null, null, false);
+
+  @Override
+  public String getName() {
+    return "StaticResourceServer";
+  }
+
+  @Override
+  public void setBindAddress(String bindAddress) {
+    this.bindAddress = bindAddress;
+  }
+
+  @Override
+  public boolean isSecure() {
+    return sslConfig.isUseSsl();
+  }
+
+  private void checkStartParams(TreeLogger logger, int port, File appRootDir) {
+    if (logger == null) {
+      throw new NullPointerException("logger cannot be null");
+    }
+
+    if (port < 0 || port > 65535) {
+      throw new IllegalArgumentException(
+              "port must be either 0 (for auto) or less than 65536");
+    }
+
+    if (appRootDir == null) {
+      throw new NullPointerException("app root directory cannot be null");
+    }
+  }
+
+  @Override
+  public boolean processArguments(TreeLogger logger, String arguments) {
+    if (arguments != null && arguments.length() > 0) {
+      Optional<SslConfiguration> parsed = SslConfiguration.parseArgs(arguments.split(","), logger);
+      if (parsed.isPresent()) {
+        sslConfig = parsed.get();
+      } else {
+        return false;
+      }
+    }
+    return true;
+  }
+
+
+  @Override
+  public ServletContainer start(TreeLogger logger, int port, File appRootDir) throws Exception {
+    TreeLogger branch = logger.branch(TreeLogger.TRACE,
+            "Starting StaticResourceServer on port " + port, null);
+
+    checkStartParams(branch, port, appRootDir);
+
+    // During startup, use the branch logger, we'll reset this later to the root logger
+    Log.setLog(new JettyTreeLogger(branch));
+
+    Server server = new Server();
+
+    ServerConnector connector = JettyLauncherUtils.getConnector(server, sslConfig, branch);
+    JettyLauncherUtils.setupConnector(connector, bindAddress, port);
+    server.addConnector(connector);
+
+    WebAppContext webAppContext = new WebAppContext(null, "/", null, null, null,
```

Comment:
Are you asking of this constructor overload is needed, or if we can pass both "/" and the path to the app dir in a different constructor? If so, my read was that we wanted `null` for these other arguments to keep Jetty from being more than a file server, and this was the most concise way to do that?

If that isn't what you're asking, can you clarify?

Reply:
Given that this can also be solved with a subdirectory in the appRootDir before the files, would you be content to leaving this kind of extra customization either to later patches, or purpose-built external jars to use for `-server`?

I can also imagine someone asking for support for h2, compression, etc, but the goal of this is to be a simple/dumb static server, rather than encourage aggressive use of it for specific use cases, as a way to get your "Hello, world!" off the ground and avoid starting a real server for your first dozen lines of code...

Reply:
For some reason I thought this already existed in the current launcher. So sure ignore that comment thread.